include/compartment.h

```diff
@@ -0,0 +1,105 @@
+#ifndef _COMPARTMENT_H
+#define _COMPARTMENT_H
+
+#include <assert.h>
+#include <elf.h>
+#include <fcntl.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/auxv.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "cheriintrin.h"
+
+#include "manager.h"
+
+#define align_down(x, align)    __builtin_align_down(x, align)
+#define align_up(x, align)      __builtin_align_up(x, align)
+
+#define SEG_MAX_COUNT 20
```

Comment:
What's this?

Reply:
This is used on line 66 of this file, since I need a static size for arrays. It's an arbitrary number for the maximum number of segments we expect to process for an ELF file. From my limited experience, the number of segments seems to be around 4, but there might be other examples.

Reply:
Got it. Can we add a comment? In the long run we probably need to make this an arbitrarily sized array, but it's not a problem for now.